ui/src/data_explorer/components/WriteDataForm.js

```diff
@@ -0,0 +1,118 @@
+import React, {Component, PropTypes} from 'react'
+
+import DatabaseDropdown from 'shared/components/DatabaseDropdown'
+import OnClickOutside from 'shared/components/OnClickOutside'
+
+class WriteDataForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedDatabase: null,
+      inputContent: '',
+    }
+
+    this.handleSelectDatabase = ::this.handleSelectDatabase
+    this.handleSubmit = ::this.handleSubmit
+    this.handleClickOutside = ::this.handleClickOutside
+    this.handleKeyUp = ::this.handleKeyUp
+    this.handleEdit = ::this.handleEdit
+  }
+
+  handleSelectDatabase(item) {
+    this.setState({selectedDatabase: item.text})
+  }
+
+  handleClickOutside(e) {
+    // guard against clicking to close error notification
+    if (e.target.className === 'overlay-technology') {
```

Comment:
I was hoping to use `ReactDOM.findDomNode(e.target)` in some kind of intelligent way, but there's no way to grab the component from that, as it reaches into the DOM. Not crazy about this, as it's brittle, but it works for now.